### ISSUE TITLE
Omniglot: Sampling classes from the same alphabet for one task

### ISSUE THREAD

**Issue by MarcCoru** [tristandeleu/pytorch-meta#25]
Hi,
First of all, thank you for developing and maintaining this repository.

I looked through the code and in particular the Omniglot dataset implementation and may have found one issue:

The `OmniglotClassDataset` defines each `alphabet/character` as separate class https://github.com/tristandeleu/pytorch-meta/blob/8e4e706c6aee07d8f6ecf8491866f18c951fcfa6/torchmeta/datasets/omniglot.py#L237

Hence, `self.num_classes` spans the number of classes times the number of alphabets.
Then the `CombinationRandomSampler` samples random indices from the entire range of classes here https://github.com/tristandeleu/pytorch-meta/blob/8e4e706c6aee07d8f6ecf8491866f18c951fcfa6/torchmeta/utils/data/sampler.py#L32

**This leads to classes from different alphabets being samples in the same task**. This arguably makes the classification task easier since characters of different alphabets may be easier to separate than characters from the same alphabet.

Is this property intended? One task should include samples from the same alphabet according to my understanding.

This may be solved by a custom `CombinationSubsetRandomSampler` that only chooses indices from samples of the same alphabet. This would require some additional alphabet-index information in the `OmniglotClassDataset` or `OmniglotDataset`.

Thanks again for creating this repository.

**Comment by tristandeleu** [comment]
Hi Marc! This was the intended behavior because I thought that you could have characters from different alphabets in the same task in the standard Omniglot dataset. For example in [Meta-Learning with Memory-Augmented Neural Networks](http://proceedings.mlr.press/v48/santoro16.pdf) they mention in Section 4.2:
> After training on 100,000 episodes with **five randomly chosen classes** with randomly chosen labels [...]

However it is true that in [Meta-Dataset](https://arxiv.org/abs/1903.03096) they do create tasks for Omniglot with characters coming exclusively from the same alphabets:
> Meta-Dataset aims to improve upon previous benchmarks in the above directions: [...] its task creation is informed by class structure for ImageNet and Omniglot

If you have reference papers that create tasks for Omniglot based on the alphabets, I'd be glad to add this!

**Comment by MarcCoru** [comment]
Hi,

Yes, you are right. It seems common practice to merge classes and alphabets as one class and sample tasks from different alphabets. This is from [Finn et al., 2017](https://arxiv.org/pdf/1703.03400.pdf)

> For Omniglot, we randomly select 1200 characters for training, irrespective of alphabet, and use the remaining for testing

Thanks for clarifying.
